tools/integration_tests_runner.py

```diff
@@ -2,136 +2,130 @@
+    process = subprocess.Popen(spell,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               shell=True
+                               )
 
-        shuffle(intermediate_tests)
-        for test in intermediate_tests:
-            self.tests.put(test)
+    out, err = process.communicate()
+    result = process.returncode
 
-        with open(self.output, "w") as self.file, open("start-finish.log", "w") as self.start_finish_log:
-            self.run_parallel_tests()
+    return filter(None, out.split("\n")), err, result
 
 
-    def run_parallel_tests(self):
-        threads = []
+def exec_test(a_tuple):
+    test_file, test, key_postfix = a_tuple
+    keys = '"--filter={test}"'.format(test=test)
 
-        for i in range(0, self.proc_count):
-            thread = Thread(target=self.exec_tests_in_queue)
-            thread.start()
-            threads.append(thread)
+    if test_file in TEMPFOLDER_TESTS:
+        tmpdir = tempfile.mkdtemp()
+        keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
+        logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
+    else:
+        keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=key_postfix)
+        logging.debug("Setting user_resource_path and data_path to {resource_path}".format(resource_path=key_postfix))
 
-        for thread in threads:
-            thread.join()
+    log_start_finish(test_file, keys, start=True)
+    out, err, result = execute_file(test_file, keys)
+    log_start_finish(test_file, keys, finish=True)
 
+    if test_file in TEMPFOLDER_TESTS:
+        try:
+            shutil.rmtree(tmpdir)
+        except:
+            logging.error("Failed to remove tempdir {tmpdir}".format(tmpdir=tmpdir))
 
-    def exec_tests_in_queue(self):
-        while not self.tests.empty():
-            try:
-                test_file, test = self.tests.get()
-                self.exec_test(test_file, test, clean_env=(test_file in TEMPFOLDER_TESTS))
+    tail = "\nEND: {file} | result: {res}\n\n".format(file=path.basename(test_file), res=result)
 
-            except:
-                logging.error(traceback.format_exc())
-                return
+    logging.getLogger(MAIN_LOGGER_NAME).info("BEGIN: {file}\n{message}{tail}".format(file=path.basename(test_file), message=str(err),  tail=tail))
+    logging.info("Finished running {test}".format(test=test))
 
 
-    def log_start_finish(self, test_file, keys, start=False, finish=False):
-        if not self.write_start_finish_log:
-            return
+def log_start_finish(test_file, keys, start=False, finish=False):
+    if (not start and not finish) or (start and finish):
+        logging.getLogger(START_FINISH_LOGGER_NAME).warning("You need to pass either start=True or finish=True, but only one of them! You passed start={0}, finish={1}".format(start, finish))
```

Comment:
So, why not to create a couple of named constants, like START and FINISH, and pass just one of them? I.e.
```python
def log_start_finish(test_file, event)
```
where event may be either START or FINISH?